app/routes/dump.py

```diff
@@ -17,9 +17,12 @@
 def requires_schema(func):
     @wraps(func)
     def wrapper(*args, **kwargs):
-        session = get_session_store()
         # pylint: disable=assigning-non-slot
-        g.schema = load_schema_from_session_data(session.session_data)
+        session_store = get_session_store()
+        metadata = get_metadata(current_user)
+        g.schema = load_schema_from_metadata(
+            metadata=metadata, language_code=session_store.session_data.language_code
```

Comment:
We should use `get_locale().language` instead of getting the session store here just for language.

Also worth updating the logic for `get_locale()` since we have started to call it from more places. It is currently doing:
```py
session = get_session_store()
return session.session_data.language_code if session else None
```

But should be updated to handle no session data.

```py
session = get_session_store()
if session and session_data := session.session_data:
	return session_data.language_code

return None
```

Reply:
Updated get_locale 👍